### PR TITLE
Remove NIO Transport Plugin

### PR DESCRIPTION
**Pull request by original-brownbear**
Removes the `transport-nio` plugin completely.

### CODE REVIEW

File: x-pack/plugin/autoscaling/build.gradle [hunk 1]
```diff
@@ -19,7 +19,6 @@ dependencies {
   testImplementation project(path: xpackModule('ilm'))
 
   testImplementation "com.fasterxml.jackson.core:jackson-databind:${versions.jackson}"
-  testImplementation project(':plugins:transport-nio') // for http
```
Comment (henningandersen) [inline x-pack/plugin/autoscaling/build.gradle:22]: Am I right in thinking that this is a left-over from #82088? Or was there never a need for this?

**Review by henningandersen — COMMENTED** [review]
Thanks Armin, just one question.

File: x-pack/plugin/autoscaling/build.gradle (hunk 1 above)
Comment (original-brownbear) [reply x-pack/plugin/autoscaling/build.gradle:22]: I don't think this was ever needed. The one test actually using HTTP `org.elasticsearch.xpack.autoscaling.action.GetAutoscalingCapacityRestCancellationIT` was always using Netty as far as I can tell from the git history. The same is true for other x-pack plugin `build.gradle` adjusted here.

**Comment by original-brownbear** [comment]
npnp :) thanks for taking a look!

**Review by henningandersen — APPROVED** [review]
LGTM.

**Comment by mark-vieira** [comment]
I feel like this should be documented in the release notes, potentially as a breaking change. We have been publishing this plugin for folks to potentially use. Might be worth some acknowledgement that it's no longer available.

**Pull request merged**